src/circular-progress.jsx

```diff
@@ -72,13 +73,14 @@ let CircularProgress = React.createClass({
     if (!this.isMounted()) return;
     if (this.props.mode !== "indeterminate") return;
 
-    wrapper.style.transform = null;
-    wrapper.style.transform = "rotate(0deg)";
+    AutoPrefix.set(wrapper.style, "transform", null);
+    AutoPrefix.set(wrapper.style, "transform", "rotate(0deg)");
     wrapper.style.transitionDuration = "0ms";
 
     setTimeout(() => {
-      wrapper.style.transform = "rotate(1800deg)";
+      AutoPrefix.set(wrapper.style, "transform", "rotate(1800deg)");
       wrapper.style.transitionDuration = "10s";
+      wrapper.style.webkitTransitionTimingFunction = "linear";
```

Comment:
@hai-cea I tested with webkit prefixed transition-duration and that did not solve the problem. The problem appears to be how Safari handles separate transition updates. Safari appears to lose the timing function. I also tested in Chrome and I did not notice any negative changes.